packages/sdk/src/cosmwasmclient.spec.ts

```diff
@@ -109,6 +111,134 @@ describe("CosmWasmClient", () => {
+          readonly tx: CosmosSdkTx;
+        }
+      | undefined;
+
+    beforeAll(async () => {
+      if (cosmosEnabled()) {
+        pendingWithoutCosmos();
+        const pen = await Secp256k1Pen.fromMnemonic(faucet.mnemonic);
+        const client = CosmWasmClient.makeReadOnly(httpUrl);
+
+        const memo = "My first contract on chain";
+        const sendMsg: MsgSend = {
+          type: "cosmos-sdk/MsgSend",
+          value: {
+            // eslint-disable-next-line @typescript-eslint/camelcase
+            from_address: faucet.address,
+            // eslint-disable-next-line @typescript-eslint/camelcase
+            to_address: makeRandomAddress(),
+            amount: [
+              {
+                denom: "ucosm",
+                amount: "1234567",
+              },
+            ],
+          },
+        };
+
+        const fee: StdFee = {
+          amount: [
+            {
+              amount: "5000",
+              denom: "ucosm",
+            },
+          ],
+          gas: "890000",
+        };
+
+        const chainId = await client.chainId();
+        const { accountNumber, sequence } = await client.getNonce(faucet.address);
+        const signBytes = makeSignBytes([sendMsg], fee, chainId, memo, accountNumber, sequence);
+        const signature = await pen.sign(signBytes);
+        const signedTx = {
+          msg: [sendMsg],
+          fee: fee,
+          memo: memo,
+          signatures: [signature],
+        };
+
+        const result = await client.postTx(marshalTx(signedTx));
+        const txDetails = await new RestClient(httpUrl).txsById(result.transactionHash);
+        posted = {
+          sender: sendMsg.value.from_address,
+          recipient: sendMsg.value.to_address,
+          hash: result.transactionHash,
+          height: Number.parseInt(txDetails.height, 10),
+          tx: txDetails.tx,
+        };
+      }
+    });
+
+    it("can search by ID", async () => {
+      pendingWithoutCosmos();
+      assert(posted, "value must be set in beforeAll()");
+      const client = CosmWasmClient.makeReadOnly(httpUrl);
+      const result = await client.searchTx({ id: posted.hash });
+      expect(result.length).toEqual(1);
+      expect(result[0]).toEqual(
+        jasmine.objectContaining({
+          height: posted.height.toString(),
+          txhash: posted.hash,
+          tx: posted.tx,
+        }),
+      );
+    });
+
+    it("can search by height", async () => {
+      pendingWithoutCosmos();
+      assert(posted, "value must be set in beforeAll()");
+      const client = CosmWasmClient.makeReadOnly(httpUrl);
+      const result = await client.searchTx({ height: posted.height });
+      expect(result.length).toEqual(1);
+      expect(result[0]).toEqual(
+        jasmine.objectContaining({
+          height: posted.height.toString(),
+          txhash: posted.hash,
+          tx: posted.tx,
+        }),
+      );
+    });
+
+    it("can search by sender", async () => {
```

Comment:
These are good cases.

I would add support for height (or min/max height) along with the sender/recipient queries.

It's not critical but I figure if you are fixing up the search implementation, it fits in.

Reason. I last queried at block 200000 and have all tx before then. I want to query for new TX, and don't want the whole list again. I can query for {sender:me, minHeight:200001}

Reply:
Unfortunately, min/max height does not work via the REST API (https://github.com/cosmos/gaia/issues/75). I made this information available in the exception text in 18aa7e33dddab20e870be18ad1a50e7e0db6afba.

I not try if height plus sender/recipient works. I guess it does. However, it would make the API much more complex and I don't see a usecase for that.